lib/presentation/widget/shared_space_details/shared_space_details_viewmodel.dart

```diff
@@ -0,0 +1,81 @@
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:domain/domain.dart';
+import 'package:linshare_flutter_app/presentation/model/shared_space_details_info.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
+import 'package:linshare_flutter_app/presentation/util/router/app_navigation.dart';
+import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
+import 'package:redux/redux.dart';
+
+class SharedSpaceDetailsViewModel extends BaseViewModel {
+  final AppNavigation _appNavigation;
+  final GetSharedSpaceInteractor _getSharedSpaceInteractor;
+  final GetQuotaInteractor _getQuotaInteractor;
+
+  SharedSpaceNodeNested sharedSpaceNodeNested;
+
+  SharedSpaceDetailsViewModel(
+    Store<AppState> store,
+    this._appNavigation,
+    this._getSharedSpaceInteractor, this._getQuotaInteractor
+  ) : super(store);
+
+  Future<SharedSpaceNodeNested> getSharedSpace(SharedSpaceId sharedSpaceId) async {
+    return (await _getSharedSpaceInteractor.execute(sharedSpaceId))
+        .map((result) => result as SharedSpaceViewState)
+        .getOrElse(() => null)
+        .sharedSpace;
+  }
+
+  Future<AccountQuota> getAccountQuota(QuotaId quotaId) async {
+    return (await _getQuotaInteractor.execute(quotaId))
+        .map((result) => result as AccountQuotaViewState)
+        .getOrElse(() => null)
+        .accountQuota;
+  }
+
+  Future<SharedSpaceDetailsInfo> getSharedSpaceDetails(SharedSpaceId sharedSpaceId) async {
+    final sharedSpace = await getSharedSpace(sharedSpaceId);
```

Comment:
I can't execute them in the same time because I need sharedSpace from the first one, but I heard you and found a solution with `then` tell me what you think 
```dart
  Future<SharedSpaceDetailsInfo> getSharedSpaceDetails(SharedSpaceDetailsArguments sharedSpaceDetailsArguments) async {
    return await getSharedSpace(sharedSpaceDetailsArguments.sharedSpaceId).then(
      (sharedSpace) => getAccountQuota(sharedSpace.quotaId).then((accountQuota) =>
        SharedSpaceDetailsInfo(sharedSpace, accountQuota)));
  }
```

Reply:
oh in that case using double `await` or `then` both good to me 👍